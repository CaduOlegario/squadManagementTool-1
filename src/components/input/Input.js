import React, {useState} from 'react';
import {Typography, TextField} from "@material-ui/core";

const Input = (props) => {
    const [isFocused, setFocused] = useState(false);

    return (
        <div>
            <Typography variant="subtitle1" color={isFocused ? "primary" : ""}>
                Nome
            </Typography>
            <TextField
                {...props}
                variant="outlined"
                onFocus={() => setFocused(true)}
                onBlur={() => setFocused(false)} />
        </div>

    )
};

export default Input;