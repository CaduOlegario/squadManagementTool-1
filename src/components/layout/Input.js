import React from 'react';
import {createMuiTheme, TextField, Typography} from "@material-ui/core";
import Box from "@material-ui/core/Box";
import {ThemeProvider} from "@material-ui/styles";
import useTheme from "@material-ui/core/styles/useTheme";

const getMuiTheme = (theme) =>
    createMuiTheme({
        ...theme,
        palette: {
            ...theme.palette,
            secondary: {
                main: 'rgb(0, 0, 0)'
            }
        },
        overrides: {
            MuiInputBase: {
                root: {
                    fontSize: "0.8rem"
                },
            },
            MuiOutlinedInput: {
                input: {
                    padding: "10px 14px"
                }
            },
        }
    });

const Input = (props) => {
    const {error} = props;
    const theme = useTheme();

    return (
        <ThemeProvider theme={getMuiTheme(theme)}>
            <Box mb={0.5}>
                <Typography variant="subtitle1"
                            color={error ? "primary" : ""} >
                    {props.label}
                </Typography>
            </Box>
            <TextField
                {...props}
                label=""
                variant="outlined"
                color="secondary"
                fullWidth
            />
        </ThemeProvider>
    )
};

export default Input;