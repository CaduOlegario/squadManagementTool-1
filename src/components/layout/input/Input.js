import React from 'react';
import {createMuiTheme, TextField, Typography} from "@material-ui/core";
import Box from "@material-ui/core/Box";
import {ThemeProvider} from "@material-ui/styles";
import useTheme from "@material-ui/core/styles/useTheme";
import PropTypes from "prop-types";

const getMuiTheme = (theme) =>
    createMuiTheme({
        ...theme,
        palette: {
            ...theme.palette,
            secondary: {
                main: 'rgb(0, 0, 0)'
            }
        },
        overrides: {
            MuiInputBase: {
                root: {
                    fontSize: "0.8rem"
                },
            },
            MuiOutlinedInput: {
                input: {
                    padding: "10px 14px"
                }
            },
        }
    });

/** Input component aligned with prototype styling.  Accepts the same props from [Material-ui text field](https://material-ui.com/components/text-fields/) */
const Input = (props) => {
    const {label, ...others} = props;
    const theme = useTheme();

    return (
        <ThemeProvider theme={getMuiTheme(theme)}>
            <Box mb={0.5}>
                <Typography variant="subtitle1"
                            color={props.error ? "primary" : undefined} >
                    {label}
                </Typography>
            </Box>
            <TextField
                {...others}
                label=""
                variant="outlined"
                color="secondary"
                fullWidth
            />
        </ThemeProvider>
    )
};

Input.propTypes ={
    /** Input title name */
    label: PropTypes.string.isRequired,
    /** Placeholder for the input */
    placeholder: PropTypes.string,
    /** Allow input to accept multiple lines */
    multiline: PropTypes.bool,
    /** Define the number of lines in the input in case its a multiline input */
    rows: PropTypes.number,
    /** Define if the component should change it's color to demonstrate an error */
    error: PropTypes.bool,
};

export default Input;