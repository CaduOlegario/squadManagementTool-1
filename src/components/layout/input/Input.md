Input example:

```js padded
<Input
    label="Nome"
    placeholder="Preencha com seu nome"
 />
```

Multiline Input Example:

```js padded
<Input 
    label="Descrição"
    placeholder="Descreva seu item"
    multiline
    rows="4"
/>
```

Input with error example:

```js padded
<Input
    label="Site"
    value="website"
    error="true"
 />
```