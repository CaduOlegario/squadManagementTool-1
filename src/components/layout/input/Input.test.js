import React from 'react';
import renderer from 'react-test-renderer';
import Input from "./Input";

it('renders correctly multiline input', () => {
    const tree = renderer
        .create(<Input
            placeholder="test placeholder"
            label="test label"
            multiline
            rows={4}
        />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly error input', () => {
    const tree = renderer
        .create(<Input
            placeholder="test placeholder"
            label="test label"
            error={true}
        />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});