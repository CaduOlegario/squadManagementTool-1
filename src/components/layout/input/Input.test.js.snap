// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly error input 1`] = `
Array [
  <div
    className="MuiBox-root MuiBox-root-6"
  >
    <h6
      className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorPrimary"
    >
      test label
    </h6>
  </div>,
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  >
    
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary Mui-error Mui-error MuiInputBase-fullWidth MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={true}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="test placeholder"
        required={false}
        type="text"
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-8"
          style={
            Object {
              "width": 0.01,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
    </div>
  </div>,
]
`;

exports[`renders correctly multiline input 1`] = `
Array [
  <div
    className="MuiBox-root MuiBox-root-1"
  >
    <h6
      className="MuiTypography-root MuiTypography-subtitle1"
    >
      test label
    </h6>
  </div>,
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  >
    
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
      onClick={[Function]}
    >
      <textarea
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="test placeholder"
        required={false}
        rows={4}
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-3"
          style={
            Object {
              "width": 0.01,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
    </div>
  </div>,
]
`;
